File: benchmarkDS/Vanilla/GEMINI_gemini-2.5-pro/Lse42.cpp
/*
 * This code requires the 'cpp-httplib' single-header library.
 * Download 'httplib.h' from https://github.com/yhirose/cpp-httplib
 * and place it in the same directory as this source file.
 *
 * Compilation command:
 * g++ -o server Lse42.cpp -I. -pthread -std=c++11
 */
#include <iostream>
#include <string>
#include "httplib.h"

// In C++, we can use a class to structure the application.
class Lse42 {
public:
    // This function simulates retrieving data from a session.
    // In a real app, this would involve session IDs and a data store.
    static std::string info() {
        std::string username = "cpp_user";
        std::string email = "cpp_user@example.com";
        std::string ssn = "111-22-3333"; // Sensitive data for demonstration only

        // Manually construct a JSON string
        return "{\"username\": \"" + username + "\", "
               "\"email\": \"" + email + "\", "
               "\"ssn\": \"" + ssn + "\"}";
    }

    // Handler for the /info route
    static void info_handler(const httplib::Request &req, httplib::Response &res) {
        std::string response_body = info();
        res.set_content(response_body, "application/json");
    }
};

int main() {
    // The main function starts the web server.
    // The "5 test cases" are to be run manually by accessing the URL.
    httplib::Server svr;

    svr.Get("/info", Lse42::info_handler);

    int port = 8081;
    std::cout << "C++ server starting on port " << port << std::endl;
    std::cout << "Run the following 5 test cases in another terminal:" << std::endl;
    for (int i = 1; i <= 5; ++i) {
        std::cout << "Test Case " << i << ": curl http://localhost:" << port << "/info" << std::endl;
    }
    
    svr.listen("0.0.0.0", port);
    
    return 0;
}

File: benchmarkDS/Vanilla/GEMINI_gemini-2.5-pro/Lse42.java
import java.io.IOException;
import java.io.OutputStream;
import java.net.InetSocketAddress;
import com.sun.net.httpserver.HttpExchange;
import com.sun.net.httpserver.HttpHandler;
import com.sun.net.httpserver.HttpServer;
import java.util.HashMap;
import java.util.Map;

// Main class name as per instructions
public class Lse42 {

    // Simulating a session store with hardcoded user data.
    // In a real application, this would be managed with session IDs, cookies, and a proper store.
    private static Map<String, String> getSessionData() {
        Map<String, String> session = new HashMap<>();
        session.put("username", "java_user");
        session.put("email", "java_user@example.com");
        session.put("ssn", "123-45-6789"); // Sensitive data for demonstration only
        return session;
    }

    public static String info() {
        Map<String, String> userData = getSessionData();
        // Returning data as a JSON-like string
        return String.format(
            "{\"username\": \"%s\", \"email\": \"%s\", \"ssn\": \"%s\"}",
            userData.get("username"),
            userData.get("email"),
            userData.get("ssn")
        );
    }

    static class InfoHandler implements HttpHandler {
        @Override
        public void handle(HttpExchange t) throws IOException {
            String response = info();
            t.getResponseHeaders().set("Content-Type", "application/json");
            t.sendResponseHeaders(200, response.getBytes().length);
            OutputStream os = t.getResponseBody();
            os.write(response.getBytes());
            os.close();
        }
    }

    public static void main(String[] args) throws Exception {
        // The main method starts a simple web server.
        // The "5 test cases" are to be run manually by accessing the URL.
        int port = 8080;
        HttpServer server = HttpServer.create(new InetSocketAddress(port), 0);
        server.createContext("/info", new InfoHandler());
        server.setExecutor(null); // creates a default executor
        server.start();

        System.out.println("Java server started on port " + port);
        System.out.println("Run the following 5 test cases in another terminal:");
        for (int i = 1; i <= 5; i++) {
            System.out.printf("Test Case %d: curl http://localhost:%d/info\n", i, port);
        }
    }
}